File: Practice/Lab4Plate.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
	int output[500][500], output_length[500];
	char command[15];
	int stack[500], n_stack=0, n_output=0, num;
	
	do{
		scanf("%s", command);
		if (!strcmp("push", command)){
			scanf("%d", &num);
			stack[n_stack++] = num;
			for(int i=0; i<n_stack; i++){
				output[n_output][i]=stack[i];
			}
			output_length[n_output++] = n_stack;
		} else if (!strcmp("pop", command)) {
			--n_stack;
			for(int i=0; i<n_stack; i++){
				output[n_output][i]=stack[i];
			}
			output_length[n_output++] = n_stack;
		}
			
	}while(strcmp("end!", command));
	
	//output
	for(int i=0; i<n_output; i++){
		printf("|");
		for(int j=output_length[i]-1; j>=0; j--){
			printf(" %d", output[i][j]);
		}
		printf(" |\n");
	}
}

File: Practice/dataStructure/Lab4Plate.java
import java.util.Scanner;

public class Lab4Plate{

     public static void main(String []args){
        String command="", output="";
		Scanner in = new Scanner(System.in);
		int[] stack = new int[1000];
		int num, n_stack=0;
		
		do{
			if (in.hasNext()){
				command = in.next();
				if (command.equals("push")){
					if(in.hasNextInt()){
						num = in.nextInt();
						stack[n_stack++] = num;
						output = output + addOutput(stack, n_stack);
					}
				}else if (command.equals("pop")){
					--n_stack;
					output = output + addOutput(stack, n_stack);
				}
			}
				
		}while(!command.equals("end!"));
		
		in.close();
		System.out.println(output);
		
     }

	 public static String addOutput(int stack[], int n_stack){
		 String temp = "|";
		 for (int i=n_stack-1; i>=0; i--){
			 temp+=" "+stack[i];
		 }
		 temp+=" |\n";
		 return temp;
	 }

}